e2e-test/cbas_azure_e2e_test.py

```diff
@@ -0,0 +1,240 @@
+
+    response = json.loads(response.text)
+    return response['run_set_id']
+
+
+# Check if outputs were written back to WDS for given record
+def check_outputs_data(wds_url, workspace_id, record_type, record_name):
+    uri = f"{wds_url}/{workspace_id}/records/v0.2/{record_type}/{record_name}"
+    headers = {"Authorization": f"Bearer {azure_token}"}
+
+    response = requests.get(uri, headers=headers)
+
+    status_code = response.status_code
+    if status_code != 200:
+        logging.error(response.text)
+        exit(1)
+    logging.info(f"Successfully retrieved record details for record '{record_name}' of type '{record_type}'")
+
+    response = json.loads(response.text)
+
+    attributes = response['attributes']
+
+    logging.info("Checking that output attributes exist in record...")
+    if 'team' in attributes and attributes['team'] == "Guardians of the Galaxy" and 'rank' in attributes and attributes['rank'] == "Captain":
+        logging.info("Outputs were successfully written back to WDS")
+    else:
+        logging.error("Outputs were not written back to WDS")
+        exit(1)
+
+
+# Check submission is in COMPLETE state
+def check_submission_status(cbas_url, method_id, run_set_id):
+    uri = f"{cbas_url}/api/batch/v1/run_sets?method_id={method_id}"
+    headers = {
+        "Authorization": f"Bearer {azure_token}",
+        "accept": "application/json"
+    }
+
+    response = requests.get(uri, headers=headers)
+    status_code = response.status_code
+
+    if status_code != 200:
+        logging.error(response.text)
+        exit(1)
+
+    response = json.loads(response.text)
+    if response['run_sets'][0]['state'] != 'COMPLETE':
+        logging.error(f"Submission '{run_set_id}' not in 'COMPLETE' state. Current state: {response['run_sets'][0]['state']}.")
+        exit(1)
+
+    logging.info(f"Submission '{run_set_id}' status: COMPLETE")
+
+
+# ---------------------- Start Workflows Azure E2E test ----------------------
+logging.info("Starting Workflows Azure E2E test...")
+
+# Create workspace
+logging.info("Creating workspace...")
+workspace_id, workspace_name = create_workspace(billing_project_name, azure_token, rawls_url)
+
+# Create WORKFLOWS_APP and CROMWELL_RUNNER apps in workspace
+create_app(workspace_id, leo_url, 'WORKFLOWS_APP', 'WORKSPACE_SHARED', azure_token)
+create_app(workspace_id, leo_url, 'CROMWELL_RUNNER_APP', 'USER_PRIVATE', azure_token)
+
+# sleep for 5 minutes to allow workspace to provision and apps to start up
+logging.info("Sleeping for 5 minutes to allow workspace to provision and apps to start up...")
+time.sleep(5 * 60)
+
+# Upload data to workspace
+# check that WDS is ready; if not exit the test after 10 minutes of polling
+logging.info(f"Checking to see if WDS app is ready to upload data for workspace {workspace_id}...")
+wds_url = poll_for_app_url(workspace_id, 'WDS', 'wds', azure_token, leo_url)
+if wds_url == "":
```

Comment:
I left it here so that the individual scripts can decide what to do when the `poll_for_app_url` returns an empty string. Here we exit the script but maybe some other script might decide that if the app url doesn't exist they might try to launch the app.